src/main/java/com/_up/megastore/controllers/interfaces/IProductController.java

```diff
@@ -18,4 +19,11 @@ ProductResponse saveProduct(
             @RequestPart MultipartFile multipartFile
     );
 
+    @PatchMapping(value = "/{productId}", consumes = { MediaType.MULTIPART_FORM_DATA_VALUE, MediaType.APPLICATION_JSON_VALUE })
+    @ResponseStatus(HttpStatus.OK)
+    ProductResponse updateProduct(
+            @RequestPart @Valid UpdateProductRequest updateProductRequest,
+            @RequestPart MultipartFile multipartFile
+    );
```

Comment:
Acá falta el parámetro con la ID del producto. La ruta está bien, puesto que vas a hacerle la petición a api/v1/products/id-del-producto. Ahora, ese ID tiene que recibirse en el método como una variable.

Debería quedarte así:

```suggestion
    @PatchMapping(value = "/{productId}", consumes = { MediaType.MULTIPART_FORM_DATA_VALUE, MediaType.APPLICATION_JSON_VALUE })
    @ResponseStatus(HttpStatus.OK)
    ProductResponse updateProduct(
            @PathVariable UUID productId,
            @RequestPart @Valid UpdateProductRequest updateProductRequest,
            @RequestPart MultipartFile multipartFile
    );
```